Dockerfile

```diff
@@ -1,3 +1,20 @@
-FROM botpress/server:v12_5_0
+FROM node:12-alpine
 WORKDIR /botpress
-CMD ["./bp"]
\ No newline at end of file
+# install dependencies
+ADD ./package.json ./yarn.lock ./
+RUN yarn install
+# add build config
+ADD ./build ./build
+ADD ./gulpfile.js ./
+# add source
+ADD ./modules ./modules
+# add botpress sdk
+ADD ./src ./src
+
+# build & package module
+# TODO extract into entry point, make it reusable to accept a list of modules
+RUN yarn run gulp build:modules --m complete-module
+RUN cd modules/complete-module && yarn && yarn build && yarn package
```

Comment:
Thinking of creating a short docker-entrypoint.sh to do these steps on demand, with arguments and more generalized.
Question: is the packaging even necessary if it gets copied directly into another botpress container?

Reply:
But you don't always package while developing it right, it's faster to test if we can skip it?

Reply:
While developing modules, we work with Botpress sources and our custom module's sources in the `modules/[your custom module]` directory. Server-side changes require a restart of the Botpress server.

Reply:
No, we do not package the module to test it locally. We simply recompile the sources (`yarn watch` inside the module directory)

Reply:
We only package the module when it's ready to be deployed in production

Reply:
So bestcase, the module builder image also exposes this watch mode so it facilitates fast module development, not only production builds...

Reply:
I think it may work like that already now, I have to do more tests and actual module development to be sure.